api/openapi.yaml

```diff
@@ -0,0 +1,209 @@
+    put:
+      tags:
+        - applications
+      summary: Adds an application
+      description: Adds an application by its address
+      operationId: addApplication
+      parameters:
+        - name: address
+          in: path
+          description: Application address
+          required: true
+          schema:
+            type: string
+            format: address
+            example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+            pattern: "^0x([0-9a-fA-F]{40})$"
+      requestBody:
+        $ref: "#/components/requestBodies/Application"
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+    delete:
+      tags:
+        - applications
+      summary: Deletes an application
+      description: delete an application
+      operationId: deleteApplication
+      parameters:
+        - name: address
+          in: path
+          description: Application address to delete
+          required: true
+          schema:
+            type: string
+            format: address
+            example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+            pattern: "^0x([0-9a-fA-F]{40})$"
+      responses:
+        "202":
+          description: Delete request accepted
+        "400":
+          description: Invalid address value
+        "404":
+          description: Application not found
+components:
+  schemas:
+    PaginatedResult:
+      type: object
+      properties:
+        total_count:
+          type: integer
+        offset:
+          type: integer
+        limit:
+          type: integer
+        data:
+          type: array
+          items: {}
+      required:
+        - total_count
+        - offset
+        - limit
+        - data
+    Application:
+      required:
+        - address
+        - blockNumber
+        - templateHash
+        - snapshotUrl
+      type: object
+      properties:
+        address:
+          type: string
+          format: address
+          example: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+          pattern: "^0x([0-9a-fA-F]{40})$"
+        blockNumber:
+          type: integer
+          example: 456311
+        templateHash:
+          type: string
+          format: hash
+          pattern: "^([0-9a-fA-F]{64})$"
+          example: "9a8d6ce861c71b592c236013b5c3fa167501431def4ad86ff59a3b12aa331dfc"
+        snapshotUrl:
+          type: string
+          format: url
+        status:
+          type: string
+          description: application status
+          enum:
+            - started
+            - starting
+            - stopping
+            - stopped
+            - error
+  requestBodies:
+    Application:
+      description: Application to be added to the node
+      required: true
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
```

Comment:
We should probably also add:
- epoch duration
- whether the app will run in reader mode

Reply:
I don't think the epoch duration whould be part of the API, for simplicity. The node would be configured (offchain) and use the same duration for all application, as this is already an offchain decision. The other params (address, block, etc) are all values that will come from the blockchain.

Reply:
How would you define the mode? An enum with values "reader", "validator" and "full". Or a "reader" boolean and a "validator" boolean.

Reply:
We won't provide a validator mode without a reader mode in Monodo because everything will be under the same database. That said, we currently use a boolean to indicate whether the node will run in reader mode. We could be consistent here. https://github.com/cartesi/rollups-node/blob/de4b8bd4d6fa86e9c6bf3e1436cafc06825ad941/internal/config/generate/Config.toml#L24

Reply:
Getting the block number might be a pain. See https://levelup.gitconnected.com/how-to-get-smart-contract-creation-block-number-7f22f8952be0

Maybe viem can come to the rescue.

Reply:
The API will be called upon a blockchain event, the block number won't be a problem.